applications/FluidDynamicsApplication/python_scripts/navier_stokes_base_solver.py

```diff
@@ -142,20 +142,20 @@ def _get_automatic_time_stepping_utility(self):
         return EstimateDeltaTimeUtility
 
     def _replace_elements_and_conditions(self):
-        ## Get element number of nodes and domain size
+        ## Get number of nodes and domain size
+        elem_num_nodes = self._get_element_num_nodes()
+        cond_num_nodes = self._get_condition_num_nodes()
         domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
-        elem_num_nodes = len(self.main_model_part.Elements.__iter__().__next__().GetNodes())
-        cond_num_nodes = len(self.main_model_part.Conditions.__iter__().__next__().GetNodes())
 
         ## Complete the element name
         if (self.element_name is not None):
-            new_elem_name = self.element_name + str(domain_size) + "D" + str(elem_num_nodes) + "N"
+            new_elem_name = self.element_name + str(int(domain_size)) + "D" + str(int(elem_num_nodes)) + "N"
```

Comment:
Just for curiosity, why here a double cast is needed?

Reply:
In a previous implementation I tried it ended up returning a double as @jcotela points. I think that the current one turns back an integer but just in case I kept the cast (in any case we can remove it).

Reply:
Not 100% how python_mpi it works underneath so if it works like this for me its ok.